Intersect.Client/Entities/Critter.cs

```diff
@@ -84,50 +84,112 @@ private void MoveRandomly()
-                            Dir = 3;
+                            Dir = Directions.Right;
                             OffsetY = 0;
                             OffsetX = -Options.TileWidth;
                         }
 
+                        break;
+                    case Directions.UpLeft:
+                        if (IsTileBlocked(X - 1, Y - 1, Z, MapId, ref blockedBy, true, true,
+                                mAttribute.IgnoreNpcAvoids) == -1 &&
+                            (!mAttribute.BlockPlayers || !PlayerOnTile(MapId, X - 1, Y - 1)))
+                        {
+                            tmpY--;
+                            tmpX--;
+                            IsMoving = true;
+                            Dir = Directions.UpLeft;
+                            OffsetY = Options.TileHeight;
+                            OffsetX = Options.TileWidth;
+                        }
+
+                        break;
+                    case Directions.UpRight:
+                        if (IsTileBlocked(X + 1, Y - 1, Z, MapId, ref blockedBy, true, true,
+                                mAttribute.IgnoreNpcAvoids) == -1 &&
+                            (!mAttribute.BlockPlayers || !PlayerOnTile(MapId, X + 1, Y - 1)))
+                        {
+                            tmpY--;
+                            tmpX++;
+                            IsMoving = true;
+                            Dir = Directions.UpRight;
+                            OffsetY = Options.TileHeight;
+                            OffsetX = -Options.TileWidth;
+                        }
+
+                        break;
+                    case Directions.DownLeft:
+                        if (IsTileBlocked(X - 1, Y + 1, Z, MapId, ref blockedBy, true, true,
+                                mAttribute.IgnoreNpcAvoids) == -1 &&
+                            (!mAttribute.BlockPlayers || !PlayerOnTile(MapId, X - 1, Y + 1)))
+                        {
+                            tmpY++;
+                            tmpX--;
+                            IsMoving = true;
+                            Dir = Directions.DownLeft;
+                            OffsetY = -Options.TileHeight;
+                            OffsetX = Options.TileWidth;
+                        }
+
+                        break;
+                    case Directions.DownRight:
+                        if (IsTileBlocked(X + 1, Y + 1, Z, MapId, ref blockedBy, true, true,
+                                mAttribute.IgnoreNpcAvoids) == -1 &&
+                            (!mAttribute.BlockPlayers || !PlayerOnTile(MapId, X + 1, Y + 1)))
+                        {
+                            tmpY++;
+                            tmpX++;
+                            IsMoving = true;
+                            Dir = Directions.DownRight;
+                            OffsetY = -Options.TileHeight;
+                            OffsetX = -Options.TileWidth;
+                        }
+
                         break;
```

Comment:
Use the switch to assign a deltaX and deltaY.

If dX/dY are both 0 after the switch, IsMoving = false and then skip the rest of the processing (because it would have fallen into the default case which isn't handled).

All of the other logic can be handled using dX/dY, we do not need to repeat the logic 8 times.